jina/drivers/evaluate.py

```diff
@@ -105,33 +108,46 @@ def extract(self, doc: 'Document') -> Any:
         return dunder_get(doc, self.field)
 
 
-class RankEvaluateDriver(FieldEvaluateDriver):
+class RankEvaluateDriver(BaseEvaluateDriver):
     """Drivers used to pass `matches` from documents and groundtruths to an executor and add the evaluation value
 
         - Example fields:
-        ['tags__id', 'id', 'score__value]
+            ['tags__id', 'score__value]
+
+    :param fields: the fields names to be extracted from the Protobuf.
+            The differences with `:class:FieldEvaluateDriver` are:
+                - More than one field is allowed. For instance, for NDCGComputation you may need to have both `ID` and `Relevance` information.
+                - The fields are extracted from the `matches` of the `Documents` and the `Groundtruth` so it returns a sequence of values.
+    :param *args:
+    :param **kwargs:
     """
 
+    @deprecated_alias(field=('fields', 0))
     def __init__(self,
-                 field: str = 'tags__id',
+                 fields: Union[str, Tuple[str]] = ('tags__id',),  # str mantained for backwards compatibility
                  *args,
                  **kwargs):
-        """
-        :param field: the field name to be extracted from the Protobuf
-        :param *args: *args for super()
-        :param **kwargs: **kwargs for super()
-        """
-        super().__init__(field, *args, **kwargs)
+        super().__init__(*args, **kwargs)
+        self.fields = fields
 
-    def extract(self, doc: 'Document'):
-        """Extract the field from the Document's matches.
+    @property
+    def single_field(self):
+        if isinstance(self.fields, str):
+            return self.fields
+        elif len(self.fields) == 1:
```

Comment:
how do you think
```python
elif isinstance(self.fields, tuple):
    # do something
```
I don't see there is a logic if else relationship between `if isinstance and elif len()`

Reply:
The problem is that some of the Executors' interfaces for backward compatibility expect a List[Union[str, float, int]], and it would not work if a Tupe of single element is an input

Reply:
okay got it!